lib/cypress/cql_bundle_importer.rb

```diff
@@ -60,114 +57,123 @@ def self.unpack_bundle(zip)
+      calc_job.execute(true)
       puts "\rLoading: Results Complete          "
     end
 
-    def self.compile_measure_relevance_hash
-      @patient_id_hash.each_value do |patient|
-        updated_patient = Patient.find(patient)
-        updated_patient.calculation_results.each do |individual_result|
-          updated_patient.update_measure_relevance_hash(individual_result)
-        end
-        updated_patient.save
-      end
-    end
-
-    def self.unpack_json(entry)
-      JSON.parse(entry.get_input_stream.read, max_nesting: false)
-    end
-
     def self.report_progress(label, percent)
       print "\rLoading: #{label} #{percent}% complete"
       STDOUT.flush
     end
 
-    def self.reconnect_valueset_references(measure)
+    def self.reconnect_valueset_references(measure, bundle)
+      # iterate over existing valueSets on measure (direct reference codes) and add bundle_id
+      measure.value_sets.each do |value_set|
+        value_set.bundle_id = bundle.id
+      end
+      # re-associate all other valueSets
       value_sets = []
-      measure['cql_libraries'].each do |cql_library|
-        cql_library['elm']['library']['valueSets'].each_pair do |_key, valuesets|
-          valuesets.each do |valueset|
-            value_sets << ValueSet.where(oid: valueset['id']).first
-          end
-        end
+      measure.cql_libraries.each do |cql_library|
+        value_sets.concat compile_value_sets_from_library(cql_library) if cql_library.elm.library['valueSets']
         next unless cql_library['elm']['library']['codes']
 
-        cql_library['elm']['library']['codes'].each_pair do |_key, codes|
-          codes.each do |code|
-            code_system_name, code_system_version = code_system_name_and_version(cql_library, code['codeSystem']['name'])
-            code_hash = ApplicationController.helpers.direct_reference_code_hash(code_system_name, code_system_version, code)
-            value_sets << ValueSet.where(oid: code_hash).first
-          end
+        value_sets.concat compile_drcs_from_library(cql_library)
+      end
+      measure.value_sets.push(*value_sets)
+    end
+
+    def self.compile_value_sets_from_library(cql_library)
+      value_sets = []
+      cql_library.elm.library.valueSets.each_pair do |_key, valuesets|
+        valuesets.each do |valueset|
+          value_sets << ValueSet.where(oid: valueset['id']).first
+        end
+      end
+      value_sets
+    end
+
+    def self.compile_drcs_from_library(cql_library)
+      value_sets = []
+      cql_library.elm.library.codes.each_pair do |_key, codes|
+        codes.each do |code|
+          code_system_name, code_system_version = code_system_name_and_version(cql_library, code['codeSystem']['name'])
+          code_hash = ApplicationController.helpers.direct_reference_code_hash(code_system_name, code_system_version, code)
+          value_sets << ValueSet.where(oid: code_hash).first
```

Comment:
If the DRC doesn't exist in the Database first, save here, and then make the reference.